extensions/text-shortcuts/src/search-shortcut.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  List,
+  LocalStorage,
+  showToast,
+  Toast,
+  Clipboard,
+  ActionPanel,
+  Action,
+  Icon,
+  useNavigation,
+  confirmAlert,
+} from "@raycast/api";
+import { useEffect, useState } from "react";
+import { fetchItemInput } from "./util/input";
+import { runShortcut, Shortcut, ShortcutSource, tags } from "./util/shortcut";
+import CreateShortcut from "./create-shortcut";
+import { preferences } from "./util/utils";
+import { ANNOTATIONS_SHORTCUTS } from "./build-in/annotation";
+import { MARKDOWNS_SHORTCUTS } from "./build-in/markdown";
+import { CASES_SHORTCUTS } from "./build-in/case";
+import { CODERS_SHORTCUTS } from "./build-in/coder";
+import { TIMES_SHORTCUTS } from "./build-in/time";
+import { FORMAT_SHORTCUTS } from "./build-in/format";
+
+export default function SearchShortcut() {
+  const [userShortcuts, setUserShortcuts] = useState<Shortcut[]>([]);
+  const [allShortcuts, setAllShortcuts] = useState<Shortcut[]>([]);
+  const [detail, setDetail] = useState<string>("");
+  const [tag, setTag] = useState<string>("All");
+  const [updateList, setUpdateList] = useState<number[]>([0]);
+  const { push } = useNavigation();
+
+  useEffect(() => {
+    async function _fetchBuildInShortcut() {
+      //user
+      const _localStorage = await LocalStorage.getItem<string>("shortcuts");
+      let _userShortcuts = [];
+      if (typeof _localStorage == "string") {
+        _userShortcuts = JSON.parse(_localStorage);
+        // setUserShortcuts(_userShortcuts);
+      }
+      //build-in
+      let _buildInShortcuts: Shortcut[] = [];
+      if (preferences().annotation) {
+        _buildInShortcuts = [..._buildInShortcuts, ...JSON.parse(ANNOTATIONS_SHORTCUTS)];
+      }
+      if (preferences().case) {
+        _buildInShortcuts = [..._buildInShortcuts, ...JSON.parse(CASES_SHORTCUTS)];
+      }
+      if (preferences().coder) {
+        _buildInShortcuts = [..._buildInShortcuts, ...JSON.parse(CODERS_SHORTCUTS)];
+      }
+      if (preferences().format) {
+        _buildInShortcuts = [..._buildInShortcuts, ...JSON.parse(FORMAT_SHORTCUTS)];
+      }
+      if (preferences().markdown) {
+        _buildInShortcuts = [..._buildInShortcuts, ...JSON.parse(MARKDOWNS_SHORTCUTS)];
+      }
+      if (preferences().time) {
+        _buildInShortcuts = [..._buildInShortcuts, ...JSON.parse(TIMES_SHORTCUTS)];
+      }
+      setAllShortcuts([..._userShortcuts.concat(_buildInShortcuts)]);
+    }
+
+    _fetchBuildInShortcut().then();
+  }, [updateList]);
+
+  return (
+    <List
+      isShowingDetail={preferences().detail}
+      isLoading={allShortcuts.length == 0}
+      searchBarPlaceholder={"Search shortcut"}
+      onSelectionChange={async (id) => {
+        const _inputItem = await fetchItemInput();
```

Comment:
hi @koinzhang - it seems like it's not very useful to call this function in the selection change handler. What about reading the data from the frontmost app/clipboard only once?

Reply:
hi @koinzhang - thanks for the explanation 🤔 let me know when you've had a chance to fix the logic in `fetchItemInput`.

Regarding the feature request: best to open an [issue](https://github.com/raycast/extensions/issues/new?assignees=&labels=api%2C+feature+request&template=api_feature_request.md&title=) so we can better discuss the use cases & track progress. 😺

Reply:
Hi, @sxn. I have updated the code, please re-review it.
I'll raise an issue to discuss about the suggestion mentioned, thanks for reminding!